src/Core/Event.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @type {Object}
+ */
+var Event = {
+	/**
+	 * @param {Element} element
+	 * @param {string} type
+	 */
+	addEventListener: function addEventListener (element, type) {
+		if (!this[type]) {
+			this[type] = new Map()
+			DOMEvent(type.substring(2), this, true)
+		}
+
+		this[type].set(DOMNode(element), element)
+	},
+	/**
+	 * @param {Event} event
+	 */
+	handleEvent: function handleEvent (event) {
```

Comment:
is `this[type]` guaranteed to be initialized here?

Reply:
Yes, an event type cannot trigger unless it has previously been added.